lib/msf/ui/console/command_dispatcher/auxiliary.rb

```diff
@@ -48,9 +55,27 @@ def method_missing(meth, *args)
 
       return mod.send(meth.to_s, *args)
     end
+
+    action = meth.to_s.delete_prefix('cmd_')
+    if mod && mod.kind_of?(Msf::Module::HasActions) && mod.actions.map(&:name).map(&:downcase).include?(action)
+       do_action(action, *args)
+    end
+
     return
   end
 
+  #
+  #
+  # Execute the module with a set action
+  #
+  def do_action(meth, *args)
+    action = mod.actions.select { |action| action.name.downcase == meth.downcase }.first
+    raise Exception('no action') if action.nil?
+    mod.datastore['ACTION'] = action.name
```

Comment:
@smcintyre-r7 , do we always want to leave the module's action set after using one of the new commands? Seems like it could be confusing to have the next `run` command repeat the most recent action run this way without the user setting it.

Reply:
That's a good point. Yeah the original should probably be restored. Should use `ensure` to ensure that's it's restored even if there's an exception raised.

Reply:
I don't think setting and resetting it is the right way to go here since `cmd_run` just uses the `mod.datastore['ACTION']` to pass the action to the `Msf::Simple` layer. I would add something like
```ruby
def cmd_run(*args, action: nil)
  action ||= mod.datastore['ACTION']
```
to `cmd_run` to ensure there's no datastore leakage.

Reply:
@mariabelenTC this implementation is really close, but you shouldn't be setting `mod.datastore['ACTION']` at all to avoid the leakage that Adam is referring to. The kwarg in `cmd_run` should be `action` instead of `cmd` like `def cmd_run(*args, action: nil)`. Then you would infer that the command is not being run with an action when `action.nil?`.

So this line 76 should be removed and L78 should be updated to be `cmd_run(*args, action: action.name)`.